src/index.js

```diff
@@ -169,51 +176,60 @@ export default class OasisApp {
   }
 
   async deviceInfo() {
-    return this.transport.send(0xe0, 0x01, 0, 0, Buffer.from([]), [0x9000, 0x6e00]).then((response) => {
-      const errorCodeData = response.slice(-2);
-      const returnCode = errorCodeData[0] * 256 + errorCodeData[1];
+    return this.transport
+      .send(0xe0, 0x01, 0, 0, Buffer.from([]), [0x9000, 0x6e00])
+      .then((response) => {
+        const errorCodeData = response.slice(-2);
+        const returnCode = errorCodeData[0] * 256 + errorCodeData[1];
+
+        if (returnCode === 0x6e00) {
+          // eslint-disable-next-line no-throw-literal
+          throw {
+            return_code: returnCode,
+            error_message: "This command is only available in the Dashboard",
+          };
+        }
+        return { response, returnCode };
+      })
+      .then(({ response, returnCode }) => {
```

Comment:
what does this new `.then` do

Reply:
I was trying to keep behavior the same, but needed to split `return error` from `return version` to fix the inferred type. Will try to find a different way

Reply:
perhaps we'd like to take this time to make a larger change that throws errors instead?

Reply:
I'd rather have that in a separate release

Reply:
(I added a helper for throwing as part of type tests https://github.com/oasisprotocol/ledger-js/pull/415/commits/c5b1493e952d049dc1103b0e2b029b9a7ff3280b )